### ISSUE TITLE
Does not recognize 4 FTDI USB DMX

### ISSUE THREAD

**Issue by ArcoSax** [OpenLightingProject/ola#1696]
I installed OLA on ub RaspBerry P3 and connected 4 FTDI USB. I have activated the relevant plugin but I only ever see 2 of the 4 connected FDTIs.

I hope not to be off topic but among the various reports I have not found (maybe I'm lazy) a solution.
Thanks for your help.

**Comment by ArcoSax** [comment]
I have to do an integration.
I bought the 3 of the 4 USB FDTUs in bulk on AliExpress, i.e. same production batch. On OLA they all have the same serial number. So when I try to associate the universe with the output it always sees only 2, the first and one of the remaining 3.
I discovered this by borrowing another FTDI from a friend and I saw that from visible 2 it went to 3.
Is there any way to make them recognize all 3 ??? Thanks!

**Comment by peternewman** [comment]
Hi @ArcoSax ,

I'm not sure off hand. If you only need three can you swap your friends one, or do you need four in total?

On the proper FTDIs I think it's possible to flash the serial, this looks like it might allow it from Linux:
http://rtr.ca/ft232r/

I've no ideal if it will work on your fake ones.

Other than that, it might be possible to do some faking with udev or hacking our code to look at USB device locations or something.

**Comment by ArcoSax** [comment]
Thanks for the link, I document and I will try to change the serial number at a low level (always if possible).

My intention was to get 4 universes using all the ports of the Raspberry.
Are there any limitations? Is the project unusable in the case of 3 or more universes?
I tested an ARTNET with ESP8266 in 2 universes but in the tests it resulted with severe delays and loss of commands.
I started making this project to test if it can stand more than 2 universes.

PS: Scusate per il linguaggio ma utilizzo google traduttore :-)

**Comment by peternewman** [comment]
> Thanks for the link, I document and I will try to change the serial number at a low level (always if possible).

Good luck.

> My intention was to get 4 universes using all the ports of the Raspberry.

Someone also added a patch so you can just use one four port interface (although that might just be in git so you'd need to build from source).

> Are there any limitations? Is the project unusable in the case of 3 or more universes?

No specific software or hardware limitations, but as with all things it will depend on processing power and what else the Pi is doing. I'd imagine it should be fine from what others have done (even on a Pi). It may also depend what model of Pi and whether you're using a hub or all the ports of the Pi. There's also the on-board UART as a possible option.

> I tested an ARTNET with ESP8266 in 2 universes but in the tests it resulted with severe delays and loss of commands.
> I started making this project to test if it can stand more than 2 universes.

Have you done a performance test with your friend's loaned device too on OLA (i.e. with 3 universes)?

> PS: Scusate per il linguaggio ma utilizzo google traduttore :-)

Il linguaggio è stato bene finora! :smiley:

**Comment by ArcoSax** [comment]
The test with ESP8266 I did them with other RS435 only chips.

If you are unable to change the serial number to the FDTI, the on-board UART solution would be interesting. where can I find information on the connections to be made ?? do i need other hardware to get?

At the moment I am using a Raspberry PI3B 2Mb but if I want I can also test everything on a PI4 4Mb.

Finally, hoping not to abuse the patience of others, somewhere I read about activating the overclock. Is it recommended or, using PI3 or PI4 it becomes superfluous ??

**Comment by peternewman** [comment]
> The test with ESP8266 I did them with other RS435 only chips.

Yeah I thought you were saying you'd got two FTDIs working already, plus your friends loaned one means you could test on a Pi with three ports, even without changing the serial number of any devices.

> If you are unable to change the serial number to the FDTI, the on-board UART solution would be interesting. where can I find information on the connections to be made ?? do i need other hardware to get?

There is detail on the UART here:
http://docs.openlighting.org/ola/conf/ola-uartdmx.conf.html

I think it will only give you one universe even on the latest Pis. You'll need an RS485 line driver of some sort, but depending on the FTDI you're using you may have added those externally already.

> At the moment I am using a Raspberry PI3B 2Mb but if I want I can also test everything on a PI4 4Mb.

I'd imagine the Pi 3B will probably be fine. They're all a massive improvement on the original Pis.

> Finally, hoping not to abuse the patience of others, somewhere I read about activating the overclock. Is it recommended or, using PI3 or PI4 it becomes superfluous ??

I think some people still do it, you're probably better off asking on the Pi forums/reading their blog. You probably want a fan if you're doing it on a Pi 4.
